core/src/test/java/org/elasticsearch/gateway/QuorumGatewayIT.java

```diff
@@ -51,72 +47,12 @@ protected int numberOfReplicas() {
         return 2;
     }
 
-    public void testChangeInitialShardsRecovery() throws Exception {
```

Comment:
can we deprecate the INDEX_RECOVERY_INITIAL_SHARDS setting? also add a breaking change note + remove docs for it? I'm fine with doing this in another PR, but we need to update the meta issue